### ISSUE TITLE
[android/premium] Automatic shutdown when all transfers finish.

### ISSUE THREAD

**Issue by gubatron** [frostwire/frostwire#192]
Request for brainstorm.

This has been requested a lot, now that we support in-app billing this should probably be one of the first premium features to implement.

How do you envision this working?

a) The app shuts down automatically as soon as all transfers finish.

b) Should the app sending a simple notification that says "All transfers finished, FrostWire exiting in 60 seconds", and there's not much you can do about it.

c) Should there be a rich notification that allows you to cancel the shutdown before it occurs?

CC: @muckachina @aldenml @marcelinkaaa @kad3mlia @alejandroarturom @votaguz

**Comment by marcelinkaaa** [comment]
For now I see it that way, if its part of paid plan:
1) upon payment there is a new entry on preferences screen - a checkbox with "Auto shutdown" after downloads are complete, it could be initially turned off. So user needs to enable it, and remember that it can be on or off. Should be on the main preference screen. 
2) if the user is downloading while not actively doing anything on the screen (the app is in the background) the app should just shut down upon download completion. If the user is downloading while actively doing something on FW app, like browsing the files, etc. we could have a notification right after activating the pro version that will remind the user that the app can be automatically shut down when the downloads are finished and its in the background. So that if they didn't check it yet they can go and enable the new setting.

**Comment by muckachina** [comment]
I agree with @marcelinkaaa suggestion (1) definitely something 'optional' because users will treat this like a bug (FW is closing by itself...) when they already have a clear 'Exit' option.

About the second point, I would also add these exceptions: 

a) When users are using FW Player, you don't want a notification or the app shutting down by itself if you are in the middle of a playlist, that would be actually a bug for a Player app.

b) Think about 'Seeding' status, you cannot turn-off the app automatically neither because is against 'Seeding'.

So there are several user case scenarios to analyze here if you want to shut-down the app when all files in transfer are totally 'Completed or Finished', it could be a complex thing to catch instead of something that simplifies the use of the app since a lot of users don't understand 'Transfer-Status' easily, specially if they don't see the differences between files from cloud or torrent sources.

**Comment by marcelinkaaa** [comment]
@muckachina 
a) I hope gubatron meant shutting down while the app is in the background only. It does not make sense any other way. And if its in the background - the user is not actively using FW. 
b) we would need a notification about seeding than for sure, so users are aware about that.

**Comment by muckachina** [comment]
In that case, we already have a notification at the top of the screen telling users FW is running in the background and people can turn it off with one click from the notification list of Android.

But thinking about 'When downloads finish' doesn't sound right for me, as a user when a download finish I don't turn-off the app, I try to open the file to check if everything is ok or listen/watch what I downloaded.

In any case If turning the app off by itself (disappearing the logo on top) I would check during a decent period of time that FW processes are not being used, files on transfer are not seeding and Player is totally off.

**Comment by gubatron** [comment]
Yes, this would only occur in the event the app is in the background and idle. Sorry for not clarifying this.

@muckachina brings a great point and it is that this feature is incompatible with Seeding. So when enabling it, if Seeding was on, the user should be warned that obviously seeding is not going to be enabled from now on, and if seeding is to be re-enabled, then the user must be warned that the auto-shutdown feature will be disabled.

**Comment by gubatron** [comment]
@muckachina "But thinking about 'When downloads finish' doesn't sound right for me"

This is basically because you're leaving a download running before going to bed. This is a common feature on other bittorrent clients, and very useful in countries where you don't have cheap electricity, etc. In fact, there's a feature to not just shutdown the app, but to shutdown the device when all downloads finish, but we're not going that far, we just want to make sure the user will not be using more data/battery.
